runtime/common/src/paras_registrar.rs

```diff
@@ -185,7 +188,15 @@ decl_module! {
 			validation_code: ValidationCode,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
-			ensure!(id >= LOWEST_USER_ID, Error::<T>::InvalidParaId);
+			let valid_id = NextFreeParaId::mutate(|id| {
+				if *id < LOWEST_USER_ID {
+					*id = LOWEST_USER_ID
+				}
+				let result = *id;
+				*id = *id + 1;
+				result
+			});
+			ensure!(id == valid_id, Error::<T>::InvalidParaId);
```

Comment:
If I read this correctly, this bumps `NextFreeParaId` slot even if the `ParaId` is occupied.
This should be fixed by having `ensure(id == max(LOWEST_USER_ID, NextFreeParaId::get() + 1))` above or something like that.

Also it's not clear to me what's the purpose of this function given there is newly introduced `register_next`? Are we worried about breaking changes? Or to give a try to occupy a given `ParaId`?

Reply:
As I wrote already:

<img width="485" alt="image" src="https://user-images.githubusercontent.com/138296/117319871-ffc94600-ae8b-11eb-89a8-95ea48663b9b.png">

Reply:
> If I read this correctly, this bumps NextFreeParaId slot even if the ParaId is occupied.

Fixed in latest logic.